pyproject.toml

```diff
@@ -36,7 +36,7 @@ dependencies = [
     "wrapt>=1.14,<2;python_version>='3.11'",
     "wrapt>=1.11,<2;python_version<'3.11'",
     "typed-ast>=1.4.0,<2.0;implementation_name=='cpython' and python_version<'3.8'",
-    "typing-extensions>=4.0.0;python_version<'3.10'",
+    "typing-extensions>=4.0.0",
```

Comment:
Why was this changed? From what I can see, typing-extensions are only required on python versions lower than 3.11 (because typing.Self is new in 3.11) and 3.13 or newer (because TypedDict will be removed from typing).

Reply:
Pipeline was failing for python 3.10 and 3.11 if we left it. https://github.com/PyCQA/astroid/actions/runs/3867121331/jobs/6591719348 / https://github.com/PyCQA/astroid/actions/runs/3867121331/jobs/6591719377

Reply:
Technically this is correct though. We should only need this on `<3.11`. However, it is easier to not have to update this and just install every time.

Are you in an environment where `typing-extensions` is unavailable?

Reply:
I tried to fix these issues in #1964 and tests pass for me locally without typing_extensions.

Reply:
> Technically this is correct though. We should only need this on `<3.11`. However, it is easier to not have to update this and just install every time.
> 
> Are you in an environment where `typing-extensions` is unavailable?

I just noticed a new typing-extensions dependency that seemed to be only necessary for tests and after looking a bit into it, it seems like it's not required for tests either with a small modification.

Reply:
I'll have a look! Thanks!